src/main/java/RegistrationService.java

```diff
@@ -0,0 +1,63 @@
+import java.util.Map;
+import java.util.Scanner;
+
+public class RegistrationService {
+    //TODO: move operations with scanner, systemState, validation, isUserLoginAvailable  to a RegistrationController
+
+    UserManagement userManagement;
+    LoginService loginService;
+
+    public RegistrationService(UserManagement userManagement, LoginService loginService) {
+        this.userManagement = userManagement;
+        this.loginService = loginService;
+    }
+
+    public void registerUser(Scanner sc, Map<String, Boolean> systemState) {
```

Comment:
Since we are using the scanner object a lot we can make it a singleton and then not pass it through functions